src/smt/smt_engine.h

```diff
@@ -152,6 +154,9 @@ class CVC4_PUBLIC SmtEngine {
   std::map< Node, std::vector< Node > > d_fmfRecFunctionsConcrete;
   NodeList* d_fmfRecFunctionsDefined;
 
+  /** Instance of the ITE remover */
+  std::unique_ptr<RemoveTermFormulas> d_iteRemover;
```

Comment:
Can this go into the SMTEnginePrivate? That would avoid the forward decl.

Reply:
There is an issue with that: I need access to `d_iteRemover` in my preprocessing pass. If it is in SmtEnginePrivate, I can't do that (without including `smt_engine.cpp` or adding a getter for d_iteRemover to SmtEngine but then we need the forward declaration again). I guess one solution would be to have a separate SmtEnginePrivate header that we could include when needed (but this should probably be a separate PR). What do you think about that?

Reply:
> I need access to d_iteRemover in my preprocessing pass.
Can you point out where this is needed and explain why it is needed? (I won't be very quick to respond so the first opportunity to discuss this might be next week.)

Reply:
I've moved `d_iteRemover` back to `SmtEnginePrivate`. The code now passes a pointer to the ITE remover to the `PreprocessingPassContext`, which is cleaner.